### PR TITLE
Exit and shield widget

### PR DESCRIPTION
**Pull request by MadWasp79**

### CODE REVIEW

File: OsmAnd/src/net/osmand/plus/routing/RouteCalculationResult.java [hunk 1]
```diff
@@ -318,16 +324,56 @@ private static void attachAlarmInfo(List<AlarmInfo> alarms, RouteSegmentResult r
...
+								}
+							}
+						}
+					}
 				}
 
-		                String description = toString(turn, ctx, false) + " " + RoutingHelper.formatStreetName(info.getStreetName(),
-						info.getRef(), info.getDestinationName(), ctx.getString(R.string.towards));
+				String description = toString(turn, ctx, false) + " " + RoutingHelper.formatStreetName(info.getStreetName(),
+						null, info.getDestinationName(), ctx.getString(R.string.towards));
```
Comment (vshcherb) [inline OsmAnd/src/net/osmand/plus/routing/RouteCalculationResult.java]: ?

File: OsmAnd-java/src/main/java/net/osmand/binary/RouteDataObject.java [hunk 2]
```diff
@@ -676,7 +676,104 @@ public boolean tunnel(){
...
+							return point[j];
+						}
+					}
+				}
+			}
+		}
+		return null;
+	}
+
+	public BinaryMapIndexReader.TagValuePair getShieldColor() {
```
Comment (vshcherb) [inline OsmAnd-java/src/main/java/net/osmand/binary/RouteDataObject.java]: ?

File: OsmAnd-java/src/main/java/net/osmand/binary/RouteDataObject.java [hunk 3]
```diff
@@ -676,7 +676,104 @@ public boolean tunnel(){
...
+				case "road_shield_color_2":
+					return new BinaryMapIndexReader.TagValuePair("road_shield_color_2", r.getValue(), 0);
+				case "road_shield_color_3":
+					return new BinaryMapIndexReader.TagValuePair("road_shield_color_3", r.getValue(), 0);
+			}
+		}
+		return null;
+	}
+
+	public BinaryMapIndexReader.TagValuePair getShieldShape() {
```
Comment (vshcherb) [inline OsmAnd-java/src/main/java/net/osmand/binary/RouteDataObject.java]: ?

File: OsmAnd/src/net/osmand/plus/routing/RouteCalculationResult.java (hunk 1 above)
Comment (MadWasp79) [reply OsmAnd/src/net/osmand/plus/routing/RouteCalculationResult.java]: fixed

File: OsmAnd-java/src/main/java/net/osmand/binary/RouteDataObject.java (hunk 3 above)
Comment (MadWasp79) [reply OsmAnd-java/src/main/java/net/osmand/binary/RouteDataObject.java]: fixed

File: OsmAnd-java/src/main/java/net/osmand/binary/RouteDataObject.java (hunk 2 above)
Comment (MadWasp79) [reply OsmAnd-java/src/main/java/net/osmand/binary/RouteDataObject.java]: fixed

**Pull request merged**